brainscore_vision/model_helpers/activations/core.py

```diff
@@ -33,16 +37,56 @@ def __init__(self, get_activations, preprocessing, identifier=False, batch_size=
         """
         :param stimuli_identifier: a stimuli identifier for the stored results file. False to disable saving.
+        :param number_of_trials: An integer that determines how many repetitions of the same model performs.
+        :param require_variance: A bool that asks models to output different responses to the same stimuli (i.e.,
+            allows stochastic responses to identical stimuli, even in otherwise deterministic base models). 
+            We here implement this using microsaccades.
+            Human microsaccade amplitude varies by who you ask, an estimate might be <0.1 deg = 360 arcsec = 6arcmin.
+            Our motivation to make use of such microsaccades is to obtain multiple different neural activities to the same input stimulus
+            from non-stochastic models. This is to improve estimates of e.g. psychophysical functions.
+            In the current implementation, if `require_variance=True`, the model microsaccades in
+            input pixel space with 1-pixel increments from the center of the stimulus. The base behavior thus
+            maintains a fixed microsaccade distance as measured in visual angle, regardless of the model's visual angle.
+            
+            Example usage:
+                `require_variance = True`
+            
+            More information:
+            --> Rolfs 2009 "Microsaccades: Small steps on a long way" Vision Research, Volume 49, Issue 20, 15
+            October 2009, Pages 2415-2441.
+            --> Haddad & Steinmann 1973 "The smallest voluntary saccade: Implications for fixation" Vision
+            Research Volume 13, Issue 6, June 1973, Pages 1075-1086, IN5-IN6.
+            Implemented by Ben Lonnqvist and Johannes Mehrer.
+
         """
+        if require_variance:
+            self.number_of_trials = number_of_trials  # for use with microsaccades
+        if (self._visual_degrees is None) and require_variance:
+            self._logger.debug("When using microsaccades for model commitments other than ModelCommitment, you should "
+                               "set self.activations_model.set_visual_degrees(visual_degrees). Not doing so risks "
+                               "breaking microsaccades.")
         if isinstance(stimuli, StimulusSet):
-            return self.from_stimulus_set(stimulus_set=stimuli, layers=layers, stimuli_identifier=stimuli_identifier)
+            return self.from_stimulus_set(stimulus_set=stimuli, layers=layers, stimuli_identifier=stimuli_identifier,
+                                          require_variance=require_variance)
         else:
-            return self.from_paths(stimuli_paths=stimuli, layers=layers, stimuli_identifier=stimuli_identifier)
+            return self.from_paths(stimuli_paths=stimuli,
+                                   layers=layers,
+                                   stimuli_identifier=stimuli_identifier,
+                                   require_variance=require_variance)
```

Comment:
maybe the following would be simpler?
```
function_call = functools.partial(self.from_stimulus_set, stimulus_set=stimuli) 
    if isinstance(stimuli, StimulusSet) else functools.partial(self.from_paths(stimuli_paths=stimuli)
return function_call(
    layers=layers, 
    stimuli_identifier=stimuli_identifier, 
    require_variance=require_variance)

Reply:
While these lines are not related to this PR per se - I see that this suggestion is simpler in the sense that it repeats less arguments. In my opinion, though, it's meaningfully more difficult to read. I've changed to:
```
        if isinstance(stimuli, StimulusSet):
            function_call = functools.partial(self.from_stimulus_set, stimulus_set=stimuli)
        else:
            function_call = functools.partial(self.from_paths, stimuli_paths=stimuli)
        return function_call(
            layers=layers,
            stimuli_identifier=stimuli_identifier,
            require_variance=require_variance)```
which I think is a nice middle ground between readability and  reduced repetition